src/sentry/release_health/release_monitor/metrics.py

```diff
@@ -0,0 +1,152 @@
+import logging
+import time
+from collections import defaultdict
+from datetime import datetime, timedelta
+from typing import Mapping, Sequence
+
+from snuba_sdk import Column, Condition, Direction, Entity, Granularity, Op, OrderBy, Query
+
+from sentry.release_health.release_monitor.base import BaseReleaseMonitorBackend, Totals
+from sentry.sentry_metrics.indexer.strings import SESSION_METRIC_NAMES
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.utils import metrics
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+
+
+class MetricReleaseMonitorBackend(BaseReleaseMonitorBackend):
+    def fetch_projects_with_recent_sessions(self) -> Mapping[int, Sequence[int]]:
+        with metrics.timer(
+            "release_monitor.fetch_projects_with_recent_sessions.loop", sample_rate=1.0
+        ):
+            aggregated_projects = defaultdict(list)
+            start_time = time.time()
+            offset = 0
+            while (time.time() - start_time) < self.MAX_SECONDS:
+                query = (
+                    Query(
+                        dataset=Dataset.Metrics.value,
+                        match=Entity(EntityKey.OrgMetricsCounters.value),
+                        select=[
+                            Column("org_id"),
+                            Column("project_id"),
+                        ],
+                        groupby=[Column("org_id"), Column("project_id")],
+                        where=[
+                            Condition(
+                                Column("timestamp"), Op.GTE, datetime.utcnow() - timedelta(hours=6)
+                            ),
+                            Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
+                            Condition(
+                                Column("metric_id"),
+                                Op.EQ,
+                                SESSION_METRIC_NAMES["c:sessions/session@none"],
```

Comment:
We usually don't access this dictionary right away but rather through the indexer in `indexer.resolve(org_id, SessionMRI.SESSION.value)` like here https://github.com/getsentry/sentry/blob/75000057b1324d301e12a0f248bf623d064a8795/src/sentry/release_health/metrics_sessions_v2.py#L587